unlock-app/src/services/web3Service.js

```diff
@@ -19,8 +19,9 @@ export const keyId = (lock, owner) => [lock, owner].join('-')
  * upstream objects.
  */
 export default class Web3Service extends EventEmitter {
-  constructor() {
+  constructor(availableProviders = providers) {
```

Comment:
```suggestion
  constructor(availableProviders) {
```

Reply:
This change will require every instance of constructing a Web3Service to pass in providers. The lockMiddleware would explicitly include it then, and call configure. Is that what we want to do? I can support this, just want to make sure that's the direction intended first.

Reply:
Ohhh you're right. Then i am confused as to why we pass `availableProviders` at all. It seems weird that we include these both via `configure(global)` above and when instantiating a new `Web3Service`. What do you think?

Reply:
The issue is that the test needs to inject an "enable" method into the HTTP provider to mock it. However, we need the rest of the provider code in order to get the stage where "connect" will actually work and call enable.  I considered injecting "enable" into the global HTTP provider, but this will literally affect all of the other unrelated tests. Since that is a dangerous anti-pattern (potentially affecting the execution of unrelated tests), I decided the best approach was to modify Web3Service to accept an injected dependency.

There are a couple of ways of doing this. The minimally invasive method is to provide a parameter, and then set its default value to the global providers, which is what I chose in order to reduce changes.

Another (and probably best) alternative is to find all the occurrences of `new Web3Service` and pass in a newly created configuration. Fortunately, in the current codebase, the only place that happens is in `src/middlewares/lockMiddleware.js`. This solution would febreze 2 code smells:

 1. use of global variable inside Web3Service
 2. hidden dependency inside lockMiddleware on the `config` global variable in Web3Service

As a consequence, this also allows us to easily mock the injected dependency for any future changes needed.

Ultimately, I think Web3Service has too many hidden dependencies and would benefit from a refactor, but that is well beyond the scope of this PR!

Reply:
Ok, I think you're right, let's just put a BIG comment above `constructor(availableProviders = providers) {` indicating why we do that (for tests...) and why this is a bad idea but that we can hopefully get rid of that code very soon as part of a big web3Service refactor. 

After that, LG!

Reply:
awesome. Thanks Julien!